### DISCUSSION TITLE
Exception: ai.djl.translate.TranslateException: This NDArray implementation does not currently support this operation

### DISCUSSION THREAD

**Discussion by zhangsongming** [deepjavalibrary/djl#2316] [Q&A]
I want to setup Djl + Paddleocr on local, but throw this exception, please help.

djl.api version: 0.15.0
java : 8
engine: onnx

**Comment by frankfliu** [comment]
OnnxRuntime engine doesn't support NDArray operations. You need install PyTorch engine as well.

**Comment by zhangsongming** [comment]
thanks for your help, but my model is onnx, so should i convert onnx model to pytorch or paddle?

**Comment by frankfliu** [comment]
You don't need convert your model. You can run your model in tensor in/tensor out mode. But if you want to do pre/post processing with NDArray operations, you need PyTorch or MXNet engine together with ONNX.

See: https://docs.djl.ai/master/docs/hybrid_engine.html
